modules/users/scripts/start_pfp_upload.ts

```diff
@@ -0,0 +1,50 @@
+import { ScriptContext, RuntimeError } from "../_gen/scripts/start_pfp_upload.ts";
+
+export interface Request {
+    mime: string;
+	contentLength: string;
+	userToken: string;
+}
+
+export interface Response {
+	url: string;
+	uploadId: string;
+}
+
+export async function run(
+	ctx: ScriptContext,
+	req: Request,
+): Promise<Response> {
+	// Authenticate/rate limit because this is a public route
+	await ctx.modules.rateLimit.throttlePublic({ period: 60, requests: 5 });
+	const { userId } = await ctx.modules.users.authenticateUser({ userToken: req.userToken });
+
+	// Ensure at least the MIME type says it is an image
+	if (!req.mime.startsWith("image/")) {
```

Comment:
We should whitelist mime types. Maybe this is something worth including in the uploads module.